src/trace_processor/perfetto_sql/generator/structured_query_generator.cc

```diff
@@ -48,6 +49,75 @@ enum QueryType : uint8_t {
   kNested,
 };
 
+std::pair<std::string, std::string> GetPreambleAndSql(const std::string& sql) {
```

Comment:
Both of these should return SqlSource *not* std::string. In general in trace processor you should avoid as much as possible leaving the world of "SqlSource".